zerver/lib/upload/__init__.py

```diff
@@ -260,14 +260,25 @@ def copy_avatar(source_profile: UserProfile, target_profile: UserProfile) -> Non
 def ensure_avatar_image(user_profile: UserProfile, medium: bool = False) -> None:
     file_path = user_avatar_path(user_profile)
 
+    final_file_path = upload_backend.get_avatar_path(file_path, medium)
+
+    if settings.LOCAL_AVATARS_DIR is not None:
+        output_path = os.path.join(
+            settings.LOCAL_AVATARS_DIR,
+            final_file_path,
+        )
+
+        if os.path.isfile(output_path):
+            return
+
```

Comment:
Before the refactoring in 9fb03cb2c7aa3a, this code to just do nothing if there exists a file at that path was used in `ensure_avatar_image`. Had to add this condition of `settings.LOCAL_AVATARS_DIR` because this is only needed for local upload backend.

Another solution could be to keep the check in `upload_single_avatar_image` but this seemed better. Also, we could just have separate functions for separate upload backend like it was before the refactor.